src/mpi/coll/transports/generic/transport.c

```diff
@@ -0,0 +1,1243 @@
+/*
```

Comment:
I think this file should be split up into two (or more files) where one handles all of the scheduling and another handles all of the vertex processing. As is, it's confusing to see all of that together. We have similar file layouts in other places where we have handlers.

Reply:
I think most of the handlers could be static functions in another file where this file would be exposing most of its functions for the collective algorithms to schedule operations.

Reply:
Ok.